File: easy/python/Question9.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        reverse = 0
        temp = x
        while x > 0:
            reverse *= 10
            reverse += x % 10
            x //= 10
        if reverse == temp:
            return True
        else:
            return False


if __name__ == "__main__":
    print(Solution().isPalindrome(-121))

File: java/easy/Question9.java
package leetcode.easy;

public class Question9 {
    public static void main(String[] args) {
        class Solution {
            public boolean isPalindrome(int x) {
                if (x < 0) {
                    return false;
                }
                int reverse = 0, temp = x;
                while (x > 0) {
                    reverse *= 10;
                    reverse += x % 10;
                    x /= 10;
                }
                if (reverse == temp) {
                    return true;
                } else {
                    return false;
                }
            }
        }

        System.out.println(new Solution().isPalindrome(1121));
    }
}